### ISSUE TITLE
Discussion: enable mix of cjs and es test code.

### ISSUE THREAD

**Issue by 3cp** [lorenzofox3/zora#155]
Following up #153, there is an easy way to enable mix of cjs and es test codes, report all of them as one.

Currently pta has to use `--module-loader` to separate cjs and es zora instance. The reason of that separation is that zora package shipped two dist files, one for cjs, one for es.

I tested following change, it all worked out:
1. don't bundle zora for es dist.
2. supply a hard-coded es dist file (may not need to be in dist folder) with following content:

```js
export {
  Assert, createHarness, createJSONReporter, createTAPReporter,
  hold, only, report, skip, test
} from './index.cjs';
```

In this way, no matter how zora is used, the same zora instance in cjs file is used. Even when you mix `test/a.spec.cjs` and `test/b.spec.mjs`, it's still the same zora instance used across cjs and es tests, hence one united report will be generated.

The other benefit is it can remove the `--module-loader` option from pta, there is no function lost, just less friction for end users.

**Comment by 3cp** [comment]
Oh, this is enough. And better for future improvements.

```js
export * from './index.cjs';
```

Reducing zora to a commonjs only package also works. (removing "exports" in package.json).
But that's a breaking change (removed `zora/cjs` and `zora/es`).

**Comment by lorenzofox3** [comment]
This sounds interesting. I am a bit concern that it implies committing to cjs where the global trend is on the contrary get rid of it. Have you tried outsied of pta ? browser, simple node module, node regular cjs package etc. 

Would you mind providing a PR so I get a better picture ?

**Comment by 3cp** [comment]
Browser is different, nodejs module resolving is not in play.
All web app bundlers will try package.json fields (by default order) "module", "browser", "main" regardless what "type" says.

(Not exactly sure about latest development of bundlers on dealing with "exports" field.)

Note "module" and "browser" are not in nodejs/npm spec, they are purely invented by browserify and webpack I think.

I used zora in browser. Pta is irrelevant in that contex. The bundler would not bundle two zora (cjs and esm). Bundler resolves `require('zora')` and `import ... from 'zora'` to same file.

I tested pta and zora in many different setups, src in cjs or esm, test in cjs or esm, all combinations work well. I have converted quite a few projects from tape or ava to zora. Really enjoy zora.

All `require()` or `import from` or `import()` can be used to consume a cjs npm package. But only `import from` or `import()` for esm npm package. That's very annoying.

In nodejs, wrapping cjs into esm is easy. The other way is unfortunately prohibited (otherwise much better choice). I'd rather Nodejs folks sided with usability rather than perfect spec conformance.

I will raise a PR.
